documentation/v&v/V&V.tex

```diff
@@ -0,0 +1,588 @@
+
+The purpose of this Verification and Validation (V\&V) is to provide a complete overview of the process undertaken in-order to ensure that all the requirements have been covered and tested. 
+\newline
+\small{ \textbf{NOTE:} Book Bazar's website where all the books to be sold are listed, is referred to as \textit{marketplace} in certain parts of this document.}
+
+\section{Scope}
+The scope for the project covers the following application features:
+\be
+	\item An authentication and authorization platform that verifies McMaster students and allows them to log into and use the application.
+	\item A marketplace system that uses information from the campus store and allows users to:
+		\bi
+			\item Post about and categorize books that they would like to sell.
+			\item Search for books that they would like to purchase.
+		\ei
+	\item A system that allows buyers and sellers to privately contact each other, agree on a price, and choose where they would like to meet.
+	\item The application will be accessible from any device with a modern browser (phone, tablet, desktop, etc.) to provide the best experience.
+\ee
+
+
+
+\section{Background}
+\label{sec:background}
+University students often like to buy/sell used textbooks. Sometimes it's for the notes written in the margins, but often it's because new textbooks are quite expensive.\\
+Currently at McMaster University, students use online websites like Facebook (\url{https://www.facebook.com/groups/macusedtexts}) or the campus store's buyback program (\url{https://campusstore.mcmaster.ca/information/guaranteed-buyback.html}, suspended because of COVID) to sell their books. These current solutions are either disorganized or take a cut of the transaction. Book Bazaar aims to create an organized marketplace that allows users to sell books for a better price than at the campus store.\\
+The project, Book Bazar is a website that addresses these issues. McMaster community members may buy/sell textbooks with ease using Book Bazar.
+\section{Roadmap}
+\label{sec: roadmap}
+Using the milestones, listed in Section 6 of the Development Process document, the team aims to have Book Bazar completed by the week of March 6, a week prior to the team's f\textbf{}inal presentation.
+
+\section{Testing Plan}
+Integration testing was used for all top-level modules which transitively tests all bottom level modules. This style of testing was chosen because it effectively tests how users will interact with the system instead of implementation details. Unit tests were not used as in practice they are often burdensome to write and maintain while providing minimal benefit in many cases due to much of the system being mocked out.
```

Comment:
We did write (one) unit test, though. I think we should change the wording to "Few unit tests were written because, in practice, they..."